docs/reference-manual/native-image/DebugInfo.md

```diff
@@ -17,6 +17,23 @@ The resulting image will contain debug records in a format the GNU Debugger (GDB
 Additionally, you can pass `-O0` to the builder which specifies that no compiler optimizations should be performed.
 Disabling all optimizations is not required, but in general it makes the debugging experience better.
 
+By default, debug info will only include details of some of the values of parameters and local variables.
+This means that the debugger will report many parameters and local variables as being undefined. If you pass `-O0` to the builder then full debug information will be included.
+If you
+want more parameter and local variable information to be included when employing higher
+levels of optimization (`-O1` or, the default, `-O2`) you need to pass an extra command
+line flag to the `native-image` command
+
+```shell
+native-image -g -H:+SourceLevelDebug Hello
+```
+
+Enabling debuginfo with flag `-H:GenerateDebugInfo=1` does not make any difference to how a generated
```

Comment:
We should use the API variant of this option. I.e. wherever we have `-H:GenerateDebugInfo=1` in `DebugInfo.md` it should be replaced with `-g`.

Reply:
Done (only two occurrences :-).